sensor/common/image/service_impl_test.go

```diff
@@ -0,0 +1,280 @@
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/internalapi/sensor"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/expiringcache/mocks"
+	"github.com/stackrox/rox/sensor/common"
+	mocks2 "github.com/stackrox/rox/sensor/common/image/mocks"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestImageService(t *testing.T) {
+	suite.Run(t, new(imageServiceSuite))
+}
+
+type imageServiceSuite struct {
+	suite.Suite
+	mockCtrl          *gomock.Controller
+	mockCache         *mocks.MockCache
+	mockRegistryStore *mocks2.MockregistryStore
+	mockCentral       *mocks2.MockcentralClient
+	mockLocalScan     *mocks2.MocklocalScan
+	service           *serviceImpl
+}
+
+var _ suite.SetupTestSuite = (*imageServiceSuite)(nil)
+
+func (suite *imageServiceSuite) SetupTest() {
+	suite.mockCtrl = gomock.NewController(suite.T())
+	suite.mockCache = mocks.NewMockCache(suite.mockCtrl)
+	suite.mockRegistryStore = mocks2.NewMockregistryStore(suite.mockCtrl)
+	suite.mockCentral = mocks2.NewMockcentralClient(suite.mockCtrl)
+	suite.mockLocalScan = mocks2.NewMocklocalScan(suite.mockCtrl)
+	suite.service = &serviceImpl{
+		imageCache:    suite.mockCache,
+		registryStore: suite.mockRegistryStore,
+		localScan:     suite.mockLocalScan,
+		centralReady:  concurrency.NewSignal(),
+		centralClient: suite.mockCentral,
+	}
+}
+
+func (suite *imageServiceSuite) TestGetImage() {
+	ctx := context.Background()
+	imageID := "imageID"
+	imageName := "imageName"
+	err := errors.New("some error")
+	errCentral := errors.Wrap(err, "scanning image via central")
+	errLocalScan := errors.Wrap(errors.New("some error"), "scanning image via local scanner")
+	cases := map[string]struct {
+		request *sensor.GetImageRequest
+		notify  common.SensorComponentEvent
+		// Unset expectValue will indicate no call to the function
+		expectCache       *expectFunctionHelper
+		expectRegistry    *expectFunctionHelper
+		expectCentralCall *expectFunctionHelper
+		expectLocalScan   *expectFunctionHelper
+		expectedError     error
+		expectedResponse  *sensor.GetImageResponse
+	}{
+		"Cache hit and central is not reachable": {
+			request:          createImageRequest(imageName, imageID, false),
+			notify:           common.SensorComponentEventOfflineMode,
+			expectCache:      expectCacheHelper(suite.mockCache, 1, createScannedImage(imageName, imageID)),
+			expectedError:    nil,
+			expectedResponse: createImageResponse(imageName, imageID),
+		},
+		"Cache hit and central is reachable": {
+			request:          createImageRequest(imageName, imageID, false),
+			notify:           common.SensorComponentEventCentralReachable,
+			expectCache:      expectCacheHelper(suite.mockCache, 1, createScannedImage(imageName, imageID)),
+			expectedError:    nil,
+			expectedResponse: createImageResponse(imageName, imageID),
+		},
+		"No cache hit and central is not reachable": {
```

Comment:
Nit
```suggestion
		"Cache miss and central is not reachable": {
```

Reply:
Also consider changing `not reachable` to `unreachable`

Reply:
...and apply to all case names in this test :)

Reply:
Done. Reads much better now :)